### DISCUSSION TITLE
Call of getControl() fails after page relod with Error: waitAsync is already running

### DISCUSSION THREAD

**Discussion by Scotti-Scholter-Marichal** [ui5-community/wdi5#347] [Q&A]
Hi everyone,

I hope this is the right section to ask quetions like these, otherwise please let me know.

I am writing E2E tests for a UI5 application with WDI5 and I am stuck trying to do a relatively simple task.
I want to reload / refresh the opened page.
Currently I am using:
`await browser.execute("location.reload(true);");`
to reload the page and afterwards I inject UI5 with the following code:
`await ui5Service.injectUI5();`

I have previously imported the ui5Service:
``const { default: _ui5Service } = require("wdio-ui5-service");
const ui5Service = new _ui5Service();``

The problem is, that if I now try to get a control with browser.getControl() after the reload, it fails with the following error:

> [0-0] [wdi5] call of _getControl() failed because of: Error: waitAsync is already running and cannot be called again at 
this moment
[0-0] [wdi5] error retrieving control: sapm.Button{"text":"Go"}

I am not sure how to fix this error and would appreciate any advise.
Thank you in advance!

Best wishes.

**Comment by Siolto** [comment]
Hi @Scotti-Scholter-Marichal,

could you maybe try to use `browser.executeAsync` instead of `browser.execute`. `execute` is a synchronous call and the executed script in the browser scope is to be assumed to be synchronous. I am not sure if `location.reload(true)` is completely synchronous and this can lead to weird side effects. WDI5 is heavily relying on the `executeAsync` scripts as well.

If the issue still occurs maybe you can add a check that the reloading really is finished and only after that try to _injectUI5_.

```javascript
await browser.executeAsync(...);
// wait till the reload is finished
await browser.$("#selectorOfTheUI5Application").waitForDisplayed()
await ui5service.injectUI5()
```

Regards,
Simon

**Reply by Scotti-Scholter-Marichal** [reply]
Hi @Siolto,

Thank you for your quick response!

If I use `await browser.executeAsync("location.reload(true);")` it throws the following error immediately after the function call:

> ERROR webdriver: Request failed with status 500 due to javascript error: javascript error: document unloaded while waiting for result

The test doesn't even reach the waiting or injectUI5 part.

Regards,
Scotti

**Reply by Siolto** [reply]
Hi @Scotti-Scholter-Marichal,

then maybe a different approach:
I had a look at the [WDIO API](https://webdriver.io/docs/api). It looks like the [`browser.url`](https://webdriver.io/docs/api/browser/url) could help us here, as it will trigger a page reload when called with the same url that is currently opened. Could you maybe try that out?

Regards,
Simon
